src/views/v3/Bridge/Routes/SingleRoute.tsx

```diff
@@ -113,21 +114,32 @@ const SingleRoute = (props: Props) => {
     isTransactionInProgress,
   });
 
-  const isHighFee = useMemo(() => {
-    if (!quote?.relayFee) {
-      return false;
+  const { isHighFee } = useMemo(() => {
+    // Compare USD value of input amount vs. output amount using a percentage threshold
+    if (!quote || !sourceToken || !destToken || !inputAmount) {
+      return { isHighFee: false };
     }
 
-    const relayFee = amount.whole(quote.relayFee.amount);
-    const feeToken = config.tokens.get(quote.relayFee.token);
-    const feePrice = calculateUSDPriceRaw(getTokenPrice, relayFee, feeToken);
+    const inputUsd = calculateUSDPriceRaw(
+      getTokenPrice,
+      inputAmount,
+      sourceToken,
+    );
+    const outputAmount = amount.whole(quote.destinationToken.amount);
+    const outputUsd = calculateUSDPriceRaw(
+      getTokenPrice,
+      outputAmount,
+      destToken,
+    );
 
-    if (feePrice === undefined) {
-      return false;
+    if (inputUsd === undefined || outputUsd === undefined || inputUsd <= 0) {
+      return { isHighFee: false };
     }
 
-    return feePrice > HIGH_FEE_THRESHOLD;
-  }, [getTokenPrice, quote?.relayFee]);
+    const delta = Math.max(0, inputUsd - outputUsd);
+    const pct = delta / inputUsd;
+    return { isHighFee: pct >= HIGH_FEE_PCT };
```

Comment:
Do we need to return an object? Or is this for future to return some other thresholds?